### ISSUE TITLE
Get rid of newBGP global variable

### ISSUE THREAD

**Issue by johananl** [metallb/metallb#578]
https://github.com/metallb/metallb/blob/9d6de1ee1b6d69a9424c4638c0fea81854b41dbb/speaker/bgp_controller.go#L289-L291

I'm assuming this global variable was created so that it can be "monkey patched" (as much as Go allows that) in tests in order to mock BGP sessions. The problem is - when this variable is overridden in a test, it affects other tests because apparently the global scope is shared among all tests.

At the moment this problem is hard to notice since we currently set this variable only twice in the tests *to the same value*:

https://github.com/metallb/metallb/blob/9d6de1ee1b6d69a9424c4638c0fea81854b41dbb/speaker/bgp_controller_test.go#L199
https://github.com/metallb/metallb/blob/9d6de1ee1b6d69a9424c4638c0fea81854b41dbb/speaker/bgp_controller_test.go#L859

However, I got bitten by this when writing a test which calls [SetConfig()](https://github.com/metallb/metallb/blob/9d6de1ee1b6d69a9424c4638c0fea81854b41dbb/speaker/bgp_controller.go#L47). Since my test ran after `newBGP` was set by a previous test, fake BGP sessions got unexpectedly added to my peers.

Unless I am missing something, I'm quite sure we can replace this abstraction with a Go interface (didn't look at this at a low level though).

**Comment by champtar** [comment]
@johananl given the man power of the project I would recommend you just do the modification and open a PR ;)

**Comment by johananl** [comment]
@champtar I'm here to help with the manpower problem :-) I've opened this issue as a reminder to myself as well as for visibility in case someone else encounters this before we fix the problem.

**Comment by champtar** [comment]
It was not 100% clear to me, then I saw another issue from you and though it was TODO

**Comment by johananl** [comment]
I guess I wasn't clear enough. Let me try and fix this :-)

I've opened this issue (#578) because I've noticed a problem with the `newBGP` variable while working on https://github.com/metallb/metallb/pull/593. I'm not working on fixing this issue (#578) at the moment. I will likely get to it eventually but there are probably more urgent issues to tackle in the project right now.

Hope that makes things clearer.
